### PR TITLE
feat: add garden planting remove function

### PR DESCRIPTION
**Pull request by williamwue**
Feature #87

### CODE REVIEW

**Comment by kyawthura18115** [comment]
Automobileengineeringservices@googlemail.com

**Comment by williamwue** [comment]
@tiembo Could you review this PR? Thx!

File: app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt [hunk 1]
```diff
@@ -16,27 +16,39 @@
...
 import org.junit.Test
+import org.junit.rules.TestRule
 
 class PlantDetailViewModelTest {
 
     private lateinit var appDatabase: AppDatabase
     private lateinit var viewModel: PlantDetailViewModel
 
+    @get:Rule
+    var rule: TestRule = InstantTaskExecutorRule()
```
Comment (XinyueZ) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt]: I don't think this `rule` is necessary needed. See comment @70

File: app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt [hunk 2]
```diff
@@ -54,4 +66,18 @@ class PlantDetailViewModelTest {
     fun testDefaultValues() {
         assertFalse(getValue(viewModel.isPlanted))
     }
+
+    @Test
```
Comment (XinyueZ) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt]: Add `@UiThreadTest`, then you could avoid the crash while testing `setValue` of  `LiveData` which couldn't be called on the thread. 

This change might reduce `InstantTaskExecutorRule` & 
```
Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java)
            // allowing main thread queries, just for testing
            .allowMainThreadQueries()
            .build()
``` above.

File: build.gradle [hunk 3]
```diff
@@ -38,6 +38,7 @@ buildscript {
         runnerVersion = '1.0.1'
         supportLibraryVersion = '28.0.0-rc01'
         workVersion = '1.0.0-alpha08'
+        coreTestingVersion = '1.1.1'
```
Comment (XinyueZ) [inline build.gradle]: Order alphabetically

**Comment by williamwue** [comment]
@tiembo No problem, I will create new PRs for this feature later

File: build.gradle (hunk 3 above)
Comment (williamwue) [reply build.gradle]: @XinyueZ Got it, I'll fix it.

File: app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt (hunk 2 above)
Comment (williamwue) [reply app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt]: @XinyueZ Good solution, Thx!

File: app/src/androidTest/java/com/google/samples/apps/sunflower/data/GardenPlantingDaoTest.kt [hunk 4]
```diff
@@ -66,6 +66,13 @@ class GardenPlantingDaoTest {
         )
     }
 
+    @Test fun testRemoveGardenPlanting() {
+        assertThat(
+            gardenPlantingDao.removeGardenPlanting(testGardenPlantingId.toString()),
+            equalTo(1)
```
Comment (saurabhkpatel) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/data/GardenPlantingDaoTest.kt]: I would use here is() matcher which is more expressive than equalTo()

Comment (saurabhkpatel) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/data/GardenPlantingDaoTest.kt]: I would use here is() matcher which is more expressive than equalTo()

File: app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt [hunk 5]
```diff
@@ -54,4 +66,18 @@ class PlantDetailViewModelTest {
...
+    fun testSnackbarMessage_AddedPlantToGardenString() {
+        assertNotEquals(viewModel.snackbarText.value, R.string.added_plant_to_garden)
```
Comment (saurabhkpatel) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt]: It's better to use assertThat over other Assert Methods.

Main Reasons:
- Readability
- Better Failure Messages
-Type Safety

Ref: https://objectpartners.com/2013/09/18/the-benefits-of-using-assertthat-over-other-assert-methods-in-unit-tests/

I'm ready to put a new PR to change in the current code base.

File: app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt [hunk 6]
```diff
@@ -54,4 +66,18 @@ class PlantDetailViewModelTest {
...
+        viewModel.showSnackbarMessage(R.string.added_plant_to_garden)
+        assertEquals(viewModel.snackbarText.value, R.string.added_plant_to_garden)
+    }
+
+    @Test
+    fun testSnackbarMessage_RemovedPlantToGardenString() {
+        assertNotEquals(viewModel.snackbarText.value, R.string.removed_plant_from_garden)
```
Comment (saurabhkpatel) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt]: Same comment here.

File: app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt [hunk 7]
```diff
@@ -54,4 +66,18 @@ class PlantDetailViewModelTest {
...
+        viewModel.showSnackbarMessage(R.string.removed_plant_from_garden)
+        assertEquals(viewModel.snackbarText.value, R.string.removed_plant_from_garden)
+    }
 }
```
Comment (saurabhkpatel) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt]: would you mind to add a new line at the end of the file?

**Pull request closed without merging**